adding support for any_w2

rupertford opened the pull request.

arporter commented:
github is reporting conflicts with master and one test fails:

    def test_dist_memory_true():
            ''' test that the distributed memory flag is on by default '''
            import config
    >       assert config.DISTRIBUTED_MEMORY
    E       assert <module 'config' from '/home/kbc59144/Projects/GungHo/PSyclone
             /src/config.pyc'>.DISTRIBUTED_MEMORY

    dynamo0p3_test.py:4136: AssertionError

Please address these issues.

arporter commented:
All modified lines in `dynamo0p3.py` are covered by the test suite. Tests have been added for fields, field vectors, operators and stencils which use the new `w2_any` space. 

Documentation has been updated appropriately. 

Examples all run OK. I don't think we need to touch those for this issue.

rupertford commented:
@arporter. I've merged master onto the branch so there are no longer conflicts. I do not get the reported test error when using pytest, either sequentially or using -n 4. Could you please check again now that the conflicts are resolved. This error was something we saw sporadically in the past and we thought it was due to using -n x and changes from one test affecting another test. I believe mock is meant to sort this problem out by ensuring any changes stay local to a test but I don't think this is associated with this pull request.

arporter commented on src/tests/test_files/dynamo0p3/testkern_anyw2_operator_mod.f90:
```diff
@@ -0,0 +1,28 @@
+!-------------------------------------------------------------------------------
```
Please correct the copyright/license information.

arporter commented:
All good now. Only minor thing is that copyright/license in one of the new files is wrong. I trust you to fix that before merging though so I'm approving this pull request :-)

rupertford commented:
Fixed the incorrect license. Renamed the test files after merging a new update from master as the new update (CMA support) used 20.* for its tests so I've moved our tests (w2any) to 21.*.
